rift/node.py

```diff
@@ -831,6 +832,9 @@ def regenerate_node_tie(self, direction, interface_going_down=None):
         self.store_tie_packet_info(packet_info)
         self.info("Regenerated node TIE for direction %s: %s",
                   packet_common.direction_str(direction), tie_packet)
+        if self.level_value() is not None:
+            self._parent_neighbors = dict(filter(lambda x: x[1].level > self.level_value(),
+                                                 tie_packet.element.node.neighbors.items()))
 
```

Comment:
Here is where parent neighbors are stored, each time the node TIE is changed (and changes in the neighbors can occur)